### ISSUE TITLE
VS Code Remote SSH not using agent to manage passwords

### ISSUE THREAD

**Issue by hgezim** [microsoft/vscode#112850]
- VSCode Version: Version: 1.52.1
Commit: ea3859d4ba2f3e577a159bc91e3074c5d85c0523
Date: 2020-12-16T16:30:02.420Z (2 days ago)
Electron: 9.3.5
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Darwin x64 19.6.0

Steps to Reproduce:

1. In a remote server, clone a GitHub repo.
2. Ensure you use an SSH key with a password for GitHub
3. Connect VS code to remote server using Remote - SSH
4. Open a foler that's checked in git
5. Go to Source Control pane and do a git pull
6. You should receive a permission denied error (expected)
7. Open a terminal (ensure it's in the remote server)
8. Run `eval "$(ssh-agent -s)"` then `ssh-add` and enter password to your default SSH key.
9. `cd <git dir>` and run `git pull`
10. That should work without issue.
11. Now go to Source Control click Pull.

Expected results:

This pull to work since `git pull` in the terminal works.

Actual results:

> git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

![Xnip2020-12-18_17-09-50](https://user-images.githubusercontent.com/124675/102674618-d9f90a00-4153-11eb-9c8a-d4d4e483f027.png)

Does this issue occur when all extensions are disabled?: Unsure.

**Comment by hgezim** [comment]
I don't believe this is a duplicate of #13680 because [comments like this](https://github.com/microsoft/vscode/issues/13680#issuecomment-660129543) suggest that using ssh-agent should fix the issue I'm having.

**Comment by roblourens** [comment]
Are you using the remote ssh-agent or expecting the local agent to be forwarded to the remote environment?
